src/renderer/contexts/SettingsContext.tsx

```diff
@@ -66,7 +66,9 @@ export const SettingsProvider = memo(
         // update theme
         const { setColorMode } = useColorMode();
         useEffect(() => {
-            setColorMode(settings.theme);
+            const [themeName, darkOrLight] = settings.theme.split('-');
+            setColorMode(darkOrLight);
+            document.documentElement.setAttribute('data-realTheme', themeName);
```

Comment:
Instead of:

```css
// Default theme (copied from Chakra UI)
:root[data-realTheme='default'] { /* ... */ }

// Charcoal theme
:root[data-theme='dark'][data-realTheme='charcoal'] { /* ... */ }
```

We could also declare themes like this:

```css
// Default theme (copied from Chakra UI)
:root { /* ... */ }

// Charcoal theme
:root[data-custom-theme='charcoal'] { /* ... */ }
```

Basically, we would use CSS declaration order to override the default colors. This should hopefully make it easier to override other colors themes might declare in the future. What do you think?

```suggestion
            const [, darkOrLight] = settings.theme.split('-');
            setColorMode(darkOrLight);
            document.documentElement.setAttribute('data-custom-theme', settings.theme);
```

Reply:
i can try